Controller/CRUDController.php

```diff
@@ -1298,6 +1274,43 @@ protected function getCsrfToken($intention)
     }
 
     /**
+     * Hook to handle access authorization.
+     *
+     * @param string $action
+     * @param object $object
+     */
+    protected function checkAccess($action, $object = null)
+    {
+        $access = array_merge(array(
+            'acl'                     => 'MASTER',
+            'export'                  => 'EXPORT',
+            'historyCompareRevisions' => 'EDIT',
+            'historyViewRevision'     => 'EDIT',
+            'history'                 => 'EDIT',
+            'edit'                    => 'EDIT',
+            'show'                    => 'VIEW',
+            'create'                  => 'CREATE',
+            'delete'                  => 'DELETE',
+            'batchDelete'             => 'DELETE',
+            'list'                    => 'LIST',
+        ), $this->admin->getAccessMapping());
+
+        if (!array_key_exists($action, $access)) {
+            throw new AccessDeniedException(sprintf('Access Denied to the action %s', $action));
```

Comment:
I think this is more an `InvalidArgumentException` than an `AccessDeniedException` and should display a message which will help developers. Something like `Action "%s" could not be found in access mapping. Please make sure your action is defined into your admin class getAccessMapping() method.`